lib/listen.js

```
@@ -3,47 +3,54 @@ var Topics = require('fh-wfm-mediator/lib/topics');
-          });
-          //TODO add catch
-      })
-      .on('update', function(object){
-          var uid = object.id;
-          self.update(object).then(function(object){
-              self.mediator.publish([self.topics.getTopic('update', 'done'), uid].join(':'), object.id);
-          });
+      .on('read', function(id) {
+        var uid = id;
+        self.read(uid).then(function(object) {
+          // TODO move topics to constants
+          self.mediator.publish([self.topics.getTopic('read', 'done'), uid].join(':'), object);
+        }).catch(function(error) {
+          console.log(error);
+        });
+        //TODO add error handling
       })
-      .on('delete', function(object){
-          var uid = object.id;
-          self.delete(object).then(function(object){
-              self.mediator.publish([self.topics.getTopic('delete', 'done'), uid].join(':'), object.id);
-          });
+      .on('update', function(object) {
+        var uid = object.id;
+        self.update(object).then(function(object) {
+          self.mediator.publish([self.topics.getTopic('update', 'done'), uid].join(':'), object.id);
+        }).catch(function(error) {
+          console.log(error);
+        });
       })
-      .on('list', function(filter){
-          self.list(filter).then(function(object){
-              self.mediator.publish(self.topics.getTopic('list', 'done'), object);
-          });
+      .on('delete', function(object) {
+        var uid = object.id;
+        self.delete(object).then(function(object) {
+          self.mediator.publish([self.topics.getTopic('delete', 'done'), uid].join(':'), object.id);
+        }).catch(function(error) {
+          console.log(error);
+        });
       })
+      .on('list', function(filter) {
+        self.list(filter).then(function(object) {
+          self.mediator.publish(self.topics.getTopic('list', 'done'), object);
+        }).catch(function(error) {
+          console.log(error);
+        });
```

If you `catch` errors and don't `return` the promise it'll just swallow the error and won't allow for `catch` calls in client code to handle the errors.

---

Moved this back to promises but still working on other issues